docker/README.md

```diff
@@ -1,39 +1,92 @@
-# Trino Gateway Docker Image
+# Trino Gateway Docker image
 
-## About the Container
+The Docker image of Trino Gateway is designed for the following use cases:
 
-This Docker image is designed to:
+* Manual usage in front of Trino clusters by mounting in a configuration file
+* Automated usage with an orchestration system like Kubernetes to simplify
+  deployment
 
-* be spun up in front of Trino clusters by mounting in a configuration file
-* simplify deployment into an orchestration system
-
-## Quickstart
-
-### Dependencies
+## Build requirements
 
 This docker build process requires:
 
 * [Docker Compose V2](https://docs.docker.com/compose/)
 * jq
 
-### Run the Trino Gateway server
+# Building a custom Docker image
+
+To build an image for a locally modified version of Trino Gateway, run the Maven
+build and then build the image for your desired processor architecture.
+
+```bash
+./build.sh arm64
+```
+
+By default, the scripts builds all valid processor architectures `amd64`,
+`arm64`, and `ppc64le`:
+
+```bash
+./build.sh
+```
+
+The Docker build process prints the ID of the image, which is also tagged with 
+`trino-gateway:xxx-SNAPSHOT-yyy`, where `xxx-SNAPSHOT` is the version number and
+`-yyy` is the processor architecture.
+
+To build an image for a specific released version of Trino Gateway, specify the 
+`-r` option. The build script downloads all the required artifacts:
+
+```bash
+./build.sh -r 4
+```
+
+Use the `-h` option for further help.
 
-You can launch the Trino Gateway and relevant dependencies through docker for testing purposes.
+### Run Trino Gateway
+
+You can launch Trino Gateway and required PostgreSQL for testing purposes with
+the following command examples.
+
+Use a locally-built image on a ARM-based machine, such as a Macbook laptop.
+
+```bash
+export TEST_GATEWAY_IMAGE="trino-gateway:6-SNAPSHOT-arm64"
+export TEST_PLATFORM="arm64"
+```
+
+Use a locally-built image on a AMD64-based machine, such as a typical Windows
+desktop or laptop.
+
+```bash
+export TEST_GATEWAY_IMAGE="trino-gateway:6-SNAPSHOT-amd64"
+export TEST_PLATFORM="amd64"
+```
+
+Use a published image for AMD64 from DockerHub.
 
 ```bash
-# Replace these variables to match your test requirements
-# Ex: You may be locally building the docker image so something like
-# `trino-gateway:5-SNAPSHOT-amd64` might be what you're expecting
+export TEST_GATEWAY_IMAGE="trinodb/trino-gateway:latest"
+export TEST_PLATFORM="amd64"
+```
+
+The release process publishes images to DockerHub for Trino Gateway 6 and newer.
+
+Set the default image
+```bash
+export TRINO_GATEWAY_IMAGE=${TEST_GATEWAY_IMAGE} 
+export DOCKER_DEFAULT_PLATFORM=${TEST_PLATFORM}
```

Comment:
note(non-blocking): i don't know if i forgot to mention this anywhere but once there's a `trinodb/trino-gateway:latest` neither of these technically need to be exported for people to spin up locally with the minimal-compose.yml file. Not sure if it's worth mentioning or not anywhere, I'll leave it to you 😄 

setting `DOCKER_DEFAULT_PLATFORM` -> Enables you to do cross platform runs when doing `docker compose`, but the default of not being set should be sufficient for local testing. We use it during the build process to test the containers as `docker compose` doesn't provide a `--platform` option

setting `TRINO_GATEWAY_IMAGE` -> Allows you to run a local image version you might've just built. (As you've already shown the readers)

Reply:
Thanks .. I have been wondering about the indirection. So from what I understand now we could just get rid of the usage of TEST_GATEWAY_IMAGE and TEST_PLATFORM just explain how to define TRINO_GATEWAY_IMAGE and DOCKER_DEFAULT_PLATFORM only .. right @rdsarvar ?

Reply:
Also .. once we published the first container to DockerHub we will separate the development docs from the docs about running it locally maybe .. or maybe even morph the docker compose setup into something suitable for spinning it up in production.. we will see

Reply:
>Thanks .. I have been wondering about the indirection. So from what I understand now we could just get rid of the usage of TEST_GATEWAY_IMAGE and TEST_PLATFORM just explain how to define TRINO_GATEWAY_IMAGE and DOCKER_DEFAULT_PLATFORM only .. right @rdsarvar ?

correct, it's only really needed in the scenario that you want to run a different architecture
ex: running arm64 and for whatever reason you want to run amd64 instead using `docker compose`

explaining how to define it makes sense 👍

one additional thing to note though: if you haven't run `build.sh` without specifying the `-a` flag you might not have pulled the different platform for the postgres image, example from the `container-test.sh` file:
```bash
    # Ensure the platform specific image has been pulled locally for Postgres
    DOCKER_DEFAULT_PLATFORM=${PLATFORM} \
        docker compose -f "${COMPOSE_PATH}" \
        pull ${COMPOSE_POSTGRES_NAME}
```

as by default if you're on arm64 you'd only have that specific image pulled, you might not have seen the errors because you ran `./docker/build.sh` which pulled all of the platform images for you


example -- 

lets run it with our arch first:
```
docker compose -f docker/minimal-compose.yml up -d
[+] Running 15/15
✔ postgres 14 layers [⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿]      0B/0B      Pulled                                                                                                               25.7s
 <LAYERS PULLED HERE>
[+] Running 3/3
✔ Network docker_default       Created                                                                                                                                      0.1s
✔ Container docker-postgres-1  Healthy                                                                                                                                      0.2s
✔ Container docker-gateway-1   Started                                                                                                                                      0.0s


docker compose -f docker/minimal-compose.yml down
[+] Running 3/3
✔ Container docker-gateway-1   Removed                                                                                                                                      0.2s
✔ Container docker-postgres-1  Removed                                                                                                                                      0.1s
✔ Network docker_default       Removed                                                                                                                                      0.1s
```

then lets run it with a different arch:
```
TRINO_GATEWAY_IMAGE="trino-gateway:6-SNAPSHOT-arm64" DOCKER_DEFAULT_PLATFORM="linux/arm64" docker compose -f docker/minimal-compose.yml up -d
[+] Running 1/0
✔ Network docker_default       Created                                                                                                                                      0.0s
⠋ Container docker-postgres-1  Creating                                                                                                                                     0.0s
Error response from daemon: image with reference postgres was found but does not match the specified platform: wanted linux/arm64, actual: linux/amd64
```
pretty much it complains because it has the postgres image locally but it's not for the right platform

so you might also need to acknowledge an explicit pull of the image:
```
DOCKER_DEFAULT_PLATFORM="linux/arm64" docker compose -f docker/minimal-compose.yml pull postgres
[+] Pulling 15/15
✔ postgres 14 layers [⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿]      0B/0B      Pulled                                                                                                               24.2s
 <LAYERS PULLED HERE>

TRINO_GATEWAY_IMAGE="trino-gateway:6-SNAPSHOT-arm64" DOCKER_DEFAULT_PLATFORM="linux/arm64" docker compose -f docker/minimal-compose.yml up -d
[+] Running 2/2
✔ Container docker-postgres-1  Healthy                                                                                                                                      0.2s
✔ Container docker-gateway-1   Started                                                                                                                                      0.0s
```

Reply:
though... the above is probably an edge case of usage? and hopefully those doing the above scenario would know to pull the image.. so i'll leave it up to you if you want to note the above information or not

Reply:
I ran into exactly that thing on macos. It complained about having linux/amd64 but needing darwin/arm64 .. so I had to delete the image and restart.

Reply:
Also .. does it make sense to set the variables each time or better to just export them once and call it a day?

Reply:
Developers could even export them in their bashrc file or whatever.

Reply:
At this stage I think we should get rid of TEST_GATEWAY_IMAGE and TEST_PLATFORM amnd just use TRINO_GATEWAY_IMAGE and DOCKER_DEFAULT_PLATFORM ..

does that make sense?

Reply:
>At this stage I think we should get rid of TEST_GATEWAY_IMAGE and TEST_PLATFORM amnd just use TRINO_GATEWAY_IMAGE and DOCKER_DEFAULT_PLATFORM ..

yup makes sense

>Also .. does it make sense to set the variables each time or better to just export them once and call it a day?

Without going through this it's hard for me to best judge what would be preferable. For now I suppose we educate them on it and let them decide?

probably not the best to set `DOCKER_DEFAULT_PLATFORM` in their shell bootup unless they're always working on that platform :thinking: